### DISCUSSION TITLE
connect wallet in react native

### DISCUSSION THREAD

**Discussion by usmanhashmis** [WalletConnect/walletconnect-monorepo#1676] [Developer Support]
`export default function Walletconnect() {
  const connector = useWalletConnect();
  
  const connectWallet = React.useCallback(() => {
    try{
       console.log("wallet");
        console.log("return",connector.connect());
        return connector.connect();
    }catch(err){
    console.log(err)
    }
     
      }, [connector]);

  const killSession = React.useCallback(() => {
    return connector.killSession();
  }, [connector]);

  
  return (
    <>
      {!connector.connected ? (
        <Button onPress={connectWallet} label="Connect a wallet" />
      ) : (
        <>
          <Text>{shortenAddress(connector.accounts[0])}</Text>
          <Button onPress={killSession} label="Log out" />
        </>
      )}
    </>
  );
}
![Screenshot 2022-11-26 103304](https://user-images.githubusercontent.com/55950977/204073882-da4c7099-2e37-4fb9-9e0f-aae4d266eb73.png)
`

When clicking a button(connect a wallet) then not open metamask wallet. connector.connect() not response any.Check screenshots

**Comment by crypblizz8** [comment]
@usmanhasmis Still need help to resolve this?
